src/Data/Unfoldable1.purs

```diff
@@ -19,6 +19,19 @@ import Partial.Unsafe (unsafePartial)
 -- | operation of a non-empty list or array; it always returns a value, and
 -- | then optionally a value to continue unfolding from.
 -- |
+-- | For example:
+-- | ``` purescript
+-- | g :: Int -> Tuple Int (Maybe Int)
+-- | g n | n <= 0 = Tuple 0 Nothing
+-- | g n = Tuple n (Just (n - 1))
+-- |
+-- | unfoldr1 g 3 == NEA.cons' 3 [2, 1, 0]
+-- | unfoldr1 g 3 == NEL.cons' 3 (2 : 1 : 0 : Nil)
+-- | -- Also produces structures of types which can be empty
+-- | unfoldr1 g 3 == [3, 2, 1, 0]
+-- | unfoldr1 g 3 == 3 : 2 : 1 : 0 : Nil
```

Comment:
Wondering if this REPL-style sequence is clearer:
```purs
> unfoldr1 g 3 :: NonEmptyArray Int
(NonEmptyArray [3,2,1,0])
> unfoldr1 g 3 :: NonEmptyList _
(NonEmptyList (NonEmpty 3 (2 : 1 : 0 : Nil)))
> unfoldr1 g 3 :: Array _
[3,2,1,0]
> unfoldr1 g 3 :: List _
(3 : 2 : 1 : 0 : Nil)
```

Also, there's some inconsistency with `show` for `NonEmptyArray` vs `NonEmptyList`. The `Array` version is a lot more concise.